### ISSUE TITLE
ctx.req.headers is undefined during export

### ISSUE THREAD

**Issue by nfriedly** [zeit/next.js#5456]
# Bug report

## Describe the bug

During execution of `next export`, in the context object passed to `getInitialProps`, ctx.req.headers` is undefined. This causes errors during export with [next-cookies](https://github.com/matthewmueller/next-cookies/blob/master/index.js#L21) (or anything else that attempts to check for any particular header).

## To Reproduce

1. clone https://github.com/nfriedly/aplexa 
2. `cd` into directory and `npm install`
3. `npm run export`

## Expected behavior

A static site is produced as if no cookies were set

## System information
 - OS: macOS 10.13.6
 - Version of Next.js: 7.0.1

**Comment by timneutkens** [comment]
https://github.com/zeit/next.js#limitation

**Comment by nfriedly** [comment]
Yea, I just found that. I still feel like this particular limitation would be a pretty straightforward fix.

**Comment by nfriedly** [comment]
Is there any explicit indication that the site is in export mode? @matthewmueller [thought](https://github.com/matthewmueller/next-cookies/pull/2#issuecomment-390034010) there was a `ctx.export` flag or something along those lines when the issue first came up over a year ago.

**Comment by nfriedly** [comment]
I'll see if I can get a PR ready, probably next week. 

I feel like there's probably a module out there with good req/res mocks already, but I'm not sure off the top of my head. Or maybe I can just instantiate "real" ones. I'll do some poking around. If nothing else looks good, then I'll just make my own basic mock.

**Comment by timneutkens** [comment]
I don't think we should mock `req`/`res` more than we do right now. It leads to unexpected behavior, we should probably remove `req` / `res` when `next export`, as it makes no sense to provide them.

**Comment by nfriedly** [comment]
I thought about that, but I expect it'll lead to more issues because there are a lot of things out there, including next-cookies, that infer that browser APIs are available if ctx.req/ctx.res is unset.

What about doing actual http requests - no mocking of anything? Then there wouldn't be 3 different environments, just the 2 that people are expecting.
